src/service.rs

```diff
@@ -599,6 +620,17 @@ where
                     Err(_toe) => {
                         if Instant::now() >= timeout_after {
                             tracing::warn!("Couldn't get vid commitment in time for block {id}",);
+                            {
+                                // we can't keep up with this block size, reduce max block size
+                                let mut global_state = self.global_state.write_arc().await;
+                                global_state.max_block_size = std::cmp::min(
+                                    global_state.max_block_size
+                                        - global_state
+                                            .max_block_size
+                                            .div_ceil(MAX_BLOCK_SIZE_CHANGE_DIVISOR),
+                                    MAX_BLOCK_SIZE_FLOOR,
+                                );
```

Comment:
IIUC `MAX_BLOCK_SIZE_FLOOR` is the minimum block size we accept, so I think this should be `cmp::max` to make sure we don't set a size lower than `MAX_BLOCK_SIZE_FLOOR`.